src/bio2bel/sources/intact.py

```diff
@@ -0,0 +1,404 @@
+
+from typing import Dict, Iterable, List
+
+import pandas as pd
+from protmapper.uniprot_client import get_mnemonic
+
+import pybel.dsl
+from bio2bel.utils import ensure_path
+from pybel import BELGraph
+
+# from ..constants import INTACT_INCREASES_ACTIONS, INTACT_DECREASES_ACTIONS, INTACT_ASSOCIATION_ACTIONS, \
+#     INTACT_BINDS_ACTIONS
+
+#: Relationship types in IntAct that map to BEL relation 'increases'
+INTACT_INCREASES_ACTIONS = {
+    'phosphorylation reaction',
+    'sumoylation reaction',
+    'methylation reaction',
+    'transglutamination reaction',
+    'ubiquitination reaction',
+    'acetylation reaction',
+    'adp ribosylation reaction',
+    'neddylation reaction',
+    'hydroxylation reaction',
+    'phosphotransfer reaction',
+    'glycosylation reaction',
+    'palmitoylation reaction',
+}
+
+#: Relationship types in IntAct that map to BEL relation 'decreases'
+INTACT_DECREASES_ACTIONS = {
+    # decreases
+    'deubiquitination reaction',
+    'protein cleavage',
+    'cleavage reaction',
+    'deacetylation reaction',
+    'lipoprotein cleavage reaction',
+    'dna cleavage',
+    'rna cleavage',
+    'dephosphorylation reaction',
+}
+
+#: Relationship types in IntAct that map to BEL relation 'association'
+INTACT_ASSOCIATION_ACTIONS = {
+    'physical association',
+    'association',
+    'colocalization',
+    'direct interaction',
+    'enzymatic reaction',
+    'atpase reaction',
+    'self interaction',
+    'gtpase reaction',
+    'putative self interaction',
+}
+
+#: Relationship types in IntAct that map to BEL relation 'hasComponent'
+INTACT_BINDS_ACTIONS = {
+    'covalent binding',
+    'disulfide bond',
+}
+
+EVIDENCE = 'From IntAct'
+SEP = '\t'
+
+MODULE_NAME = 'intact'
+URL = 'ftp://ftp.ebi.ac.uk/pub/databases/intact/current/psimitab/intact.zip'
+path = ensure_path(MODULE_NAME, URL)
+sample_path = '/Users/sophiakrix/Downloads/intact_sample.txt'
+
+ID_INTA = '#ID(s) interactor A'
+ID_INTB = 'ID(s) interactor B'
+INTERACTION_TYPES = 'Interaction Type(s)'
+PUBLICATION_ID = 'Publication Identifier(s)'
+DATABASE_INT_A = 'database_intA'
+DATABASE_INT_B = 'database_intB'
+ONLY_ID_INT_A = 'id_intA'
+ONLY_ID_INT_B = 'id_intB'
+UNIPROTKB = 'uniprotkb'
+SOURCE = 'source'
+TARGET = 'target'
+RELATION = 'relation'
+PUBMED_ID = 'pubmed_id'
+columns_mapping = {
+    '#ID(s) interactor A': SOURCE,
+    'ID(s) interactor B': TARGET,
+    'Interaction Type(s)': RELATION,
+    'Publication Identifier(s)': PUBMED_ID,
+}
+
+
+def _load_file(module_name: str = MODULE_NAME, url: str = URL) -> str:
+    """Load the file from the URL and place it into the bio2bel_sophia directory.
+
+    :param module_name: name of module (database)
+    :param url: URL to file from database
+    :return: path of saved database file
+    """
+    return ensure_path(prefix=module_name, url=url)
+
+
+def _get_my_df() -> pd.DataFrame:
+    """Get my dataframe."""
+    path = _load_file()
```

Comment:
this zip file has many files in it. try:

```python
from zipfile import ZipFile

import pandas as pd

path = ensure_path(...)
with ZipFile(path) as zip_file:
        with zip_file.open('intact.txt') as file:
            return pd.read_csv(file, sep='\t')
```

Reply:
the other file is `intact_negative.txt` do you know what that is? Is it information about what interactions don't happen? That would be an awesome use case for https://github.com/pybel/pybel/pull/408

Reply:
I only receive an intact.txt file, nothing else. Where do you receive a folder?

Reply:
the URL you put downloads a zip file. it looks like they also distribute the full text file separately. Your call. Downloading/storing zipped data seems better IMO

Reply:
Works now: df3ecf7e3b583be01ffaae886a37b79a49763605